src/main/java/uk/gov/hmcts/reform/sandl.snlevents/service/FactMessageService.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.sandl.snlevents.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlevents.model.Problem;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.ProblemRepository;
+
+import java.io.IOException;
+
+@Service
+public class FactMessageService {
```

Comment:
Not sure if I follow you with this class and usage, but we can talk about that tomorrow.

Reply:
atm it's class that will handle the responses from the rules engine with new facts added. shortly this will be triggered by the eventing system so it's kind of already prepared for it.